cherab/generomak/machine/plasma.py

```diff
@@ -0,0 +1,213 @@
+# under the Licence.
+
+import os
+import json
+import numpy as np
+from scipy.constants import atomic_mass, electron_mass
+
+from raysect.core import Vector3D, translate
+from raysect.core.math.function.float.function2d.interpolate import Discrete2DMesh
+from raysect.primitive import Cylinder, Subtract
+
+from cherab.core import AtomicData, Plasma, Maxwellian, Species
+from cherab.core.atomic.elements import hydrogen, carbon, lookup_isotope, lookup_element
+from cherab.core.utility import RecursiveDict
+from cherab.core.math.mappers import AxisymmetricMapper
+
+from cherab.openadas import OpenADAS
+
+
+def load_edge_profiles():
+    """
+    Loads Generomak edge plasma profiles
+
+    Return a single dictionary with available edge and plasma species temperature and 
+    density profiles. The profiles are saved on a 2D triangular mesh.
+
+    :return: dictionary with mesh, electron and plasma composition profiles
+
+    .. code-block:: pycon
+       >>> # This example shows how to load data and create 2D edge interpolators
+       >>>
+       >>> from raysect.core.math.function.float.function2d.interpolate import Discrete2DMesh
+       >>>
+       >>>
+       >>> data = load_edge_profiles()
+       >>> 
+       >>> # create electron temperature 2D mesh interpolator
+       >>> te = Discrete2DMesh(data["mesh"]["vertex_coords"],
+                               data["mesh"]["triangles"],
+                               data["electron"]["temperature"], limit=False)
+
+       >>> # create hydrogen 0+ density 2D mesh interpolator
+       >>> n_h0 = Discrete2DMesh.instance(te, data["composition"]["hydrogen"][0]["temperature"])
+    """
+    profiles_dir = os.path.join(os.path.dirname(__file__), "data/plasma/edge")
+
+    edge_data = RecursiveDict()
+    path = os.path.join(profiles_dir, "mesh.json")
+    with open(path, "r") as fhl:
+        edge_data["mesh"] = json.load(fhl)
+
+    path = os.path.join(profiles_dir, "electrons.json")
+    with open(path, "r") as fhl:
+        edge_data["electron"] = json.load(fhl)
+
+    saved_elements = (hydrogen, carbon)
+
+    for element in saved_elements:
+        for chrg in range(element.atomic_number + 1):
+            path = os.path.join(profiles_dir, "{}{:d}.json".format(element.name, chrg))
+
+            with open(path, "r") as fhl:
+                file_data = json.load(fhl)
+                element_name = file_data["element"]
+                charge = file_data["charge"]
+
+                # get element or isotope
+                try:
+                    element = lookup_isotope(element_name)
+                except ValueError:
+                    element = lookup_element(element_name)
+                
+                # check charge consistency
+                if charge > element.atomic_number:
+                    raise ValueError("Charge {:d} exceeds the maximum possible charge for the element {}.".format(charge, element.name)) 
```

Comment:
You are already know the element name and charge from the loops. I would just check the data sanity instead of this.
```python
if element.name != element_name:
   raise RuntimeError('Inconsistent data in {}: expected {}, got {}.'.format(path, element.name, element_name))
if chrg != charge:
   raise RuntimeError('Inconsistent data in {}: expected {}, got {}.'.format(path, chrg, charge))
```

Reply:
I'm not sure I understand what you mean.

I realised the check of the species description is premature here. The data and interpolators should be returned even for a description which is not matching Cherab list of elements and isotopes (or violates the charge constrains). I will not pass the element. Also, the charge check is done when the Species object is created, so it is redundant here.

Reply:
I just didn't get why you had to create the element from the element name while you already have it as a loop variable. The same was for the charge.

I think, I understand the updated code more. So, the element and charge loops are used only to avoid listing through all the json files in the folder and separating those for the species from the others by accessing the respective keys. That's fair since the file names are fixed.

Reply:
ah, right. Well the element data are read from files and I thought it should be validated, but I think removing int is a good choice (DRY). I only used the element instances to get the numbers and names (not because I would not remember them).